drivers/gpio/gpio_nrfx.c

```diff
@@ -171,7 +197,18 @@ static int gpio_nrfx_pin_configure(const struct device *port, gpio_pin_t pin,
 		__ASSERT_NO_MSG(err == NRFX_SUCCESS);
 	}
 
-	return 0;
+end:
+#ifdef CONFIG_SOC_NRF54H20_GPD
+	if (cfg->pad_pd == NRF_GPD_FAST_ACTIVE1) {
+		nrf_gpio_port_retain_set(cfg->port, retain);
```

Comment:
`| BIT(pin)` seems missing, so the retention won't be applied to the pin we would like to apply